src/hn/ui/MainWindow.ui

```diff
@@ -1,56 +1,47 @@
 <?xml version="1.0" encoding="UTF-8"?>
-<!-- Generated with glade 3.22.2 -->
 <interface>
-  <requires lib="gtk+" version="3.20"/>
-  <requires lib="libhandy" version="1.0"/>
-  <template class="AppWindow" parent="HdyApplicationWindow">
-    <property name="can_focus">False</property>
-    <property name="default_width">360</property>
-    <property name="default_height">720</property>
+  <requires lib="gtk" version="4.0"/>
+  <requires lib="libadwaita" version="1.0"/>
+  <template class="AppWindow" parent="AdwApplicationWindow">
     <child>
       <object class="GtkBox">
         <property name="orientation">vertical</property>
         <child>
-          <object class="HdyDeck" id="header_stack">
-            <property name="visible">True</property>
-            <property name="transition_type">slide</property>
-            <property name="hexpand">True</property>
+          <object class="AdwLeaflet" id="header_stack">
+            <property name="can-unfold">False</property>
             <child>
-              <object class="NewsHeader"></object>
+              <object class="NewsHeader"/>
             </child>
             <child>
-              <object class="ThreadHeader"></object>
+              <object class="ThreadHeader"/>
             </child>
             <child>
-              <object class="WebsiteHeader"></object>
+              <object class="WebsiteHeader"/>
             </child>
           </object>
         </child>
         <child>
-          <object class="HdyDeck" id="stack">
-            <property name="visible">True</property>
-            <property name="can_focus">False</property>
-            <property name="can_swipe_back">True</property>
-            <property name="visible_child">news_list</property>
-            <property name="transition_type">slide</property>
+          <object class="AdwLeaflet" id="stack">
+            <property name="can-unfold">False</property>
+            <property name="vexpand">True</property>
             <child>
-              <object class="NewsList" id="news_list"></object>
+              <object class="NewsList" id="news_list"/>
             </child>
             <child>
-              <object class="CommentThread" id="ct"></object>
+              <object class="CommentThread" id="ct"/>
             </child>
             <child>
-              <object class="WebsiteView" id="www"></object>
+              <object class="WebsiteView" id="www"/>
             </child>
           </object>
         </child>
       </object>
     </child>
   </template>
-  <object class="HdySwipeGroup">
+  <object class="AdwSwipeGroup">
```

Comment:
This has been removed. Use one `AdwLeaflet` - you probably want to combine the `*Header` files with their respective views.